client/flutter/lib/api/iso_country.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:core';
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+import 'package:yaml/yaml.dart';
+
+class IsoCountryList {
+  Map<String, IsoCountry> countries;
+
+  static const isoFilePath = 'assets/onboarding/iso_countries.en.yaml';
+
+  static final IsoCountryList _singleton = IsoCountryList._internal();
+
+  factory IsoCountryList() {
+    return _singleton;
+  }
+
+  IsoCountryList._internal();
+
+  static Future<Map<String, IsoCountry>> countriesFromYaml(
+      String csvPath) async {
+    Map<String, IsoCountry> countries = {};
+    final yamlString = await rootBundle.loadString(IsoCountryList.isoFilePath);
+    final yaml = loadYaml(yamlString);
+    yaml['countries'].forEach((countryYaml) {
+      final country = IsoCountry(
+        name: countryYaml['name'],
+        alpha2Code: countryYaml['alpha_2_code'],
+      );
+      countries[country.alpha2Code] = country;
+    });
+    return countries;
+  }
+
+  Future<bool> loadCountries() async {
+    if (countries != null && countries.isNotEmpty) {
```

Comment:
is it possible to have a race cond here

Reply:
2 things to address this:
* Since `countries` is only really intended to be set once, I can make it a private variable with a read-only getter so that nothing else alters it
* I don't think I need the `isNotEmpty` here... I was thinking about it as a potential way to rebound from an error by re-fetching the list, since if there are no countries, the user can't get past onboarding, but that's not the best way to handle. I'll remove that call

Reply:
Side note, shorthand syntax would be `countries?.isNotEmpty ?? false`

Reply:
Updated and applied the shorthand (did not remove the `isNotEmpty` for now)